precompiles/werc20/integration_test.go

```diff
@@ -0,0 +1,539 @@
+
+	Expect(erc20Balance.String()).To(Equal(balanceAfter.Balance.Amount.BigInt().String()), "expected return balance from contract equal to bank")
+	return erc20Balance
+}
+
+// -------------------------------------------------------------------------------------------------
+// Integration tests
+// -------------------------------------------------------------------------------------------------
+
+var _ = When("a user interact with the WEVMOS precompiled contract", func() {
+	var (
+		// callArgs factory.CallArgs
+		// txArgs   evmtypes.EvmTxArgs
+
+		passCheck, failCheck, transferCheck, depositCheck, withdrawCheck testutil.LogCheckArgs
+
+		callsData CallsData
+
+		txSender, user keyring.Key
+	)
+
+	depositAmount := big.NewInt(1e18)
+	withdrawAmount := depositAmount
+	transferAmount := depositAmount
+
+	BeforeEach(func() {
+		is.SetupTest()
+
+		precompileAddr := common.HexToAddress(is.precompileAddrHex)
+		// TODO: is the denom correct for both mainnet and tesnet?
+		tokenPair := erc20types.NewTokenPair(
+			precompileAddr,
+			evmtypes.GetEVMCoinDenom(),
+			erc20types.OWNER_MODULE,
+		)
+
+		precompile, err := werc20.NewPrecompile(
+			tokenPair,
+			is.network.App.BankKeeper,
+			is.network.App.AuthzKeeper,
+			is.network.App.TransferKeeper,
+		)
+		Expect(err).ToNot(HaveOccurred(), "failed to instantiate the werc20 precompile")
+		is.precompile = precompile
+
+		failCheck = testutil.LogCheckArgs{ABIEvents: is.precompile.Events}
+		passCheck = failCheck.WithExpPass(true)
+
+		withdrawCheck = passCheck.WithExpEvents(werc20.EventTypeWithdrawal)
+		depositCheck = passCheck.WithExpEvents(werc20.EventTypeDeposit)
+		transferCheck = passCheck.WithExpEvents(erc20.EventTypeTransfer)
+
+		txSender = is.keyring.GetKey(0)
+		user = is.keyring.GetKey(1)
+
+		callsData = CallsData{
+			sender: txSender,
+
+			precompileAddr: precompileAddr,
+			precompileABI:  precompile.ABI,
+		}
+	})
+	Context("calling a specific wrapped coin method", func() {
+		Context("and funds are part of the transaction", func() {
+			When("the method is deposit", func() {
+				It("it should return funds to sender and emit the event", func() {
+					// Store initial balance to verify that sender
+					// balance remains the same after the contract call.
+					initBalance := is.checkAndReturnBalance(passCheck, callsData, user.Addr)
+					Expect(is.network.NextBlock()).ToNot(HaveOccurred(), "error on NextBlock")
+
+					txArgs, callArgs := callsData.getTxAndCallArgs(directCall, werc20.DepositMethod)
+					txArgs.Amount = depositAmount
+
+					_, _, err := is.factory.CallContractAndCheckLogs(callsData.sender.Priv, txArgs, callArgs, depositCheck)
+					Expect(err).ToNot(HaveOccurred(), "unexpected error calling the precompile")
+					Expect(is.network.NextBlock()).ToNot(HaveOccurred(), "error on NextBlock")
+
+					finalBalance := is.checkAndReturnBalance(passCheck, callsData, user.Addr)
+					Expect(finalBalance).To(Equal(initBalance))
```

Comment:
Shouldnt it be `initBalance - fee`?

Reply:
The setup is intentionally made to have zero fees transactions. Since fee computation is not part of the logic involved here, without them it is easier to tests and to read the tests. Do you have any particular test you wish to see with non zero gas price?